cppapi/client/devapi_attr.tpp

```diff
@@ -77,9 +77,12 @@ void DeviceAttribute::base_val(T _val)
 	d_state_filled = false;
 	exceptions_flags.set(failed_flag);
 	exceptions_flags.set(isempty_flag);
+	// This constructor is supposed to be used only for enum types since all
+	// the constructors for the other standard Tango types have a specialization defined
```

Comment:
In this case we should remove template from here and add standard overloading. Should be clear then, should not be?

Reply:
How would you support the case where _val is of type enum Color, or enum MyEnum?
This is why this method is here.